triton/examples/whisper/whisper.rst

```diff
@@ -0,0 +1,141 @@
+is not necessary, but whisper's language detection is sometimes weird. 
+
+If you are transcribing language different 
+from English, use a general model e.g. ``$medium``. If your source 
+audio is in English, using English-specific models is usually a 
+performance gain.
+
+For full list of options, run:
+
+.. code-block:: bash
+
+   singularity_wrapper run --help
+
+Running on GPU
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Singularity-wrapper takes care of making GPUs available for the container, 
+so all you need to do to run Whisper on a GPU is use the previous 
+command and add additional flag: ``--device cuda``. 
+Without this, Whisper will only run on a CPU even if a GPU is availabe.
+
+Usage (Whisper-diarization)
+------------------------------------
+
+This example shows you a sample script to run whisper-diarization.
+
+.. code-block:: bash
+
+    module load whisper_diarization
+    srun --mem=6G singularity_wrapper run -a your_audio_file.wav --whisper-model $medium_en
+
+Option ``--whisper-model $medium_en`` Tells whisper which model to use, in this case 
+``medium.en``. If you use environment variables that come with the module to specify the 
+model, whisper will run using a local model. Otherwise it will download the model to 
+your home directory. For list of all local models, run ``echo $model_names`` with 
+whisper-diarization loaded.
+
+Note that syntax is unfortunately somewhat different compared to plain whisper. You 
+need to specify the audio file to use with the argument ``-a audio_file.wav`` and 
+similarily the syntax to specificy the model is different.
+
+For full list of options, run:
+
+.. code-block:: bash
+
+   singularity_wrapper run --help
+
+Running on GPU
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Compared to plain Whisper, running whisper-diarization on GPU takes little 
+more work. Singularity-wrapper still takes care of making GPUs available 
+for the container and you still specify you want to use GPU using the flag 
+``--device cuda``. 
+
+Unfortunately whisper-diarization requires multiple models when using a GPU
+, and there isn't a practical way to use local models for this. For this 
+reason, you should create a symlink from whisper's cache folder in your 
+home, to your work directory. This way you avoid filling your home 
+directory's quota.
+
+To do this, run following commands:
+
+.. code-block:: bash
+    
+    mkdir -p {~/.cache/huggingface/,~/.cache/torch/NeMo,temp_cache/huggingface/,temp_cache/NeMo/,$WRKDIR/whisper_cache/huggingface,$WRKDIR/whisper_cache/NeMo}
+    mv ~/.cache/huggingface/* temp_cache/huggingface/
+    mv ~/.cache/torch/NeMo/* temp_cache/NeMo/
+    rmdir {~/.cache/huggingface/,~/.cache/torch/NeMo}
+    ln -s $WRKDIR/whisper_cache/huggingface ~/.cache/
+    ln -s $WRKDIR/whisper_cache/NeMo ~/.cache/torch/
+    mv temp_cache/huggingface/* ~/.cache/huggingface/ 
+    mv temp_cache/NeMo/* ~/.cache/torch/NeMo
+    rmdir {temp_cache/huggingface,temp_cache/NeMo,temp_cache}
```

Comment:
```suggestion
.. code-block:: console
    
    $ mkdir -p ~/.cache/huggingface/ ~/.cache/torch/NeMo temp_cache/huggingface/ temp_cache/NeMo/ $WRKDIR/whisper_cache/huggingface $WRKDIR/whisper_cache/NeMo
    $ mv ~/.cache/huggingface/* temp_cache/huggingface/
    $ mv ~/.cache/torch/NeMo/* temp_cache/NeMo/
    $ rmdir ~/.cache/huggingface/ ~/.cache/torch/NeMo
    $ ln -s $WRKDIR/whisper_cache/huggingface ~/.cache/
    $ ln -s $WRKDIR/whisper_cache/NeMo ~/.cache/torch/
    $ mv temp_cache/huggingface/* ~/.cache/huggingface/
    $ mv temp_cache/NeMo/* ~/.cache/torch/NeMo
    $ rmdir temp_cache/huggingface temp_cache/NeMo temp_cache
```
The `{}` shouldn't be needed with full arguments (but it would be useful in cases like this: `rmdir temp_cache/{huggingface,NeMo}`)

Reply:
Does this work if both it already exists, and if it doesn't already exist?  After reading carefully I can't see anything wrong, but it is rather complex stuff to do...